### PR TITLE
Sg/2024 01 24b

### PR DESCRIPTION
**Pull request by docuracy**
Added spatial pre-filter on ccodes using Select2.

Note addition of `twitter-typeahead-components-bundle/main.js` in CDN fallbacks and check that it has been found when rebuilding webpack.

Ignore comment "Refactored country_feature_collection as a utility function" - I meant `regions_countries`, but that is now a hard-coded read of a JSON file.

### CODE REVIEW

File: utils/regions_countries.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    # regions = [{'id': region['id'], 'text': region['title'], 'ccodes': region['ccodes']} for region in sorted(regions, key=lambda x: x['title'])]
+    # countries = [{'id': country['ccodes'][0], 'text': country['title']} for country in sorted(countries, key=lambda x: x['title'])]
+    #
+    # dropdown_data = [{'text':'Regions', 'children': list(regions)}, {'text': 'Countries', 'children': list(countries)}]
+    
+    # # Save the regenerated data to the JSON file
+    # with open('static/js/regions_countries.json', 'w') as json_file:
+    #     json.dump(dropdown_data, json_file, default=str)
+    
+    with open('static/js/regions_countries.json', 'r') as file:
```
Comment (kgeographer) [inline utils/regions_countries.py:38]: this file in static is not tracked - media/data is where several static files have lived typically

Comment (docuracy) [reply utils/regions_countries.py:38]: Ah, OK - will you move it and update the code or shall I?

Comment (kgeographer) [reply utils/regions_countries.py:38]: I don't have it, as static is not tracked - please send it and I'll include it

**Comment by docuracy** [comment]
Ah yes, sorry... Ta-da!

File: utils/regions_countries.py (hunk 1 above)
Comment (docuracy) [reply utils/regions_countries.py:38]: [regions_countries.json](https://github.com/WorldHistoricalGazetteer/whg3/files/14075511/regions_countries.json)

**Pull request merged**